src/goto-programs/class_hierarchy.cpp

```diff
@@ -50,6 +50,8 @@ void class_hierarchy_grapht::populate(const symbol_tablet &symbol_table)
         const irep_idt &parent = to_symbol_type(base.type()).get_identifier();
         if(!parent.empty())
         {
+          DATA_INVARIANT(nodes_by_name.find(parent)!=nodes_by_name.end(),
+                         "parent class not in symbol table");
```

Comment:
The subsequent code uses `at`, hence there would be an exception anyway. I guess the added value lies in the error message? It would be nice to do `const auto node_it = nodes_by_name.find(parent);` and use `node_it` in the `DATA_INVARIANT` as well as `add_edge` below.

Reply:
ok, done
yes, the value-add is the better diagnostics